File: Newton/src/Newton/utilities.clj
(ns Newton.utilities)

;; To keep same name conventions as utilities.lisp
;; In idiomatic clojure, this could be replaced by the anonymous function
;; \#(apply + %)
(defn sum
  "Sums the elements of a list or vector"
  [list]
    (apply + list))

File: Newton/utilities.jl
# CDT Utilities
#
# Various utility functions ported from cdt-utilities.lisp
# List operations in Lisp are replaced by tuple or 1-d array operations
# in Julia.
# Only dimension independent utilities are in this file
# Many functions such as standard deviation and mean are built into Julia already,
# and hence do not need to be defined here.

function sum(list)
	# Sums the elements in a tuple or 1-d array
	return apply(+, list)
end

